### DISCUSSION TITLE
Best Way to Avoid "TypeError: Cannot read property "PID.13.1" from undefined?"

### DISCUSSION THREAD

**Discussion by flugruger** [nextgenhealthcare/connect#5042] [Q&A]
What's the best way to avoid errors from empty fields & components?
I'm hoping for something short, not a full try/catch. 

What's the simplest way to avoid this in Mirth?

**Comment by jonbartels** [comment]
https://mirthconnect.slack.com/archives/C02SW0K4D/p1644579455116479 - Check out this thread in Slack. It is from 2022-02-11. It should remain in Slack history for a few days. The user "vill" had the same question.

Please help others by copying the pieces you find useful from Slack to this Discussion to preserve them.

**Reply by flugruger** [reply]
Thanks, @jonbartels ... looking at that Slack conversation, I see these ideas:

- msg['PID']['PID.18']['PID.18.1'].toString().length > 0  (from Jay)
- msg['PID']['PID.18']['PID.18.1'].length() > 0                (from Agermano)
- if ('PID.18.1' in msg['PID']['PID.18']) {}                        (from Agermano)
- input == null || input.trim().length == 0                    (from CreepySheep)

I'll mess with these and likely put one into a function. 

Thanks!

**Reply by tonygermano** [reply]
The method Jay posted will tell you if the field is not Empty. The two I shared will tell you if the field is defined (empty or not.)

If you are getting the error in your description, you are probably trying to access a (potentially) repeating field or segment by index, and the index is not valid for your particular message. That's pretty much the only way to get `undefined` for an xml object.

**Reply by flugruger** [reply]
> If you are getting the error in your description, you are probably trying to access a (potentially) repeating field or segment by index, and the index is not valid for your particular message. That's pretty much the only way to get `undefined` for an xml object.

@tonygermano ... I think you're right. The code was something like this:
	```tmp['phone']    = msg['PID']['PID.13'][0]['PID.13.1'].toString().replace(/[^\d]/g, '');```

But I feel like this has happened in other situations. If I find an alternate example, I'll post it.
